api/src/services/user-service.ts

```diff
@@ -36,22 +48,26 @@ export class UserService {
 		};
 	};
 
-	getUser = async (identityId: string): Promise<User | null> => {
+	getUser = async (identityId: string, isAuthorizedUser?: boolean): Promise<User | null> => {
 		const userPersistence = await userDb.getUser(identityId);
-		return userPersistence && this.getUserObject(userPersistence);
-	};
+		const user = userPersistence && this.getUserObject(userPersistence);
+		const privateUserInfo: boolean = user?.isPrivate && !isAuthorizedUser;
 
-	getUsersByIds = async (identityIds: string[]): Promise<User[] | null> => {
-		const usersPersistence = await userDb.getUsersByIds(identityIds);
-		if (!usersPersistence) {
+		if (!user) {
 			return null;
 		}
-		return usersPersistence.map((userP: UserPersistence) => this.getUserObject(userP));
+
+		return {
+			...user,
+			claim: !privateUserInfo ? user.claim : undefined,
```

Comment:
Same here, you can just delete `user.claim`

Reply:
I don't know how this should work in the return object creation, I would need another line having an if check.